From a16a1a9dabd65cd129b625a65459487f49227cae Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Mon, 23 Mar 2015 16:53:33 +0800
Subject: [PATCH] mmc: card: disable SW CMDQ properly for the host which
 doesn't support

SW CMDQ is not disabled properly if a host doesn't support SW CMDQ. This
patch will fix this

Change-Id: I013fee434104ae8bcd265f1a2af2dac9435953e7
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-27131
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/card/block.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 6a0ba38..8242868 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -713,7 +713,8 @@ static inline int mmc_blk_part_switch(struct mmc_card *card,
 			pm_suspend_ignore_children(&host->class_dev, true);
 		} else if ((md->part_type == EXT_CSD_PART_CONFIG_USER) &&
 					card->ext_csd.cmdq_support &&
-				!card->ext_csd.cmdq_en) {
+					(host->caps2 & MMC_CAP2_CAN_DO_CMDQ) &&
+					!card->ext_csd.cmdq_en) {
 			/* enable CMDQ mode */
 			ret = mmc_switch(card, EXT_CSD_CMD_SET_NORMAL,
 					EXT_CSD_CMDQ_MODE_EN,
-- 
1.9.1

